### PR TITLE
[stylelint-plugin] feat: migrate from @import to @use statements

### PR DESCRIPTION
**Pull request by kmark**
`@blueprintjs/stylelint-plugin`  now supports Sass `@use` and drops support for Sass `@import`. Less `@import` support remains.

### CODE REVIEW

File: packages/stylelint-plugin/README.md [hunk 1]
```diff
@@ -52,10 +52,10 @@ Enforce usage of the color variables instead of color literals.
 -.my-class {
 -    border: 1px solid #137CBD;
 -}
-+ @import "~@blueprintjs/core/lib/scss/variables";
++ @use "@blueprintjs/core/lib/scss/variables" as bp;
```
Comment (styu) [inline packages/stylelint-plugin/README.md]:
```suggestion
+ @use "@blueprintjs/core/lib/scss/variables.scss" as bp;
```

File: packages/stylelint-plugin/README.md [hunk 2]
```diff
@@ -83,9 +83,9 @@ The `@blueprintjs` package exports a `bp-ns` CSS variable which contains the pre
 -.bp3-button > div {
 -    border: 1px solid black;
 -}
-+ @import "~@blueprintjs/core/lib/scss/variables";
++ @use "@blueprintjs/core/lib/scss/variables" as bp;
```
Comment (styu) [inline packages/stylelint-plugin/README.md]:
```suggestion
+ @use "@blueprintjs/core/lib/scss/variables.scss" as bp;
```

File: packages/stylelint-plugin/README.md [hunk 3]
```diff
@@ -83,9 +83,9 @@ The `@blueprintjs` package exports a `bp-ns` CSS variable which contains the pre
...
 +
-+.#{$bp-ns}-button > div {
++.#{bp.$bp-ns}-button > div {
```
Comment (styu) [inline packages/stylelint-plugin/README.md]:
```suggestion
+.#{bp.$ns}-button > div {
```

File: packages/stylelint-plugin/src/utils/cssSyntax.ts [hunk 4]
```diff
@@ -29,13 +29,15 @@ export const BpVariablePrefixMap: Record<Exclude<CssSyntax, CssSyntax.OTHER>, st
     [CssSyntax.LESS]: "@",
 };
 
+export const BpSassNamespace = "bp";
+
 export const BpPrefixVariableMap: Record<Exclude<CssSyntax, CssSyntax.OTHER>, string> = {
-    [CssSyntax.SASS]: "#{$bp-ns}",
+    [CssSyntax.SASS]: `#{${BpSassNamespace}.$bp-ns}`,
```
Comment (styu) [inline packages/stylelint-plugin/src/utils/cssSyntax.ts]:
```suggestion
    [CssSyntax.SASS]: `#{${BpSassNamespace}.$ns}`,
```

File: packages/stylelint-plugin/test/no-prefix-literal.test.js [hunk 5]
```diff
@@ -198,8 +198,8 @@ describe("no-prefix-literal", () => {
             expect(warnings).lengthOf(0);
 
             const fixedSourceContents = fs.readFileSync(mutableFixturePath, { encoding: "utf-8" });
-            expect(fixedSourceContents).to.contain(`@import "~@blueprintjs/core/lib/scss/variables";`);
-            expect(fixedSourceContents).to.contain(".#{$bp-ns}-tag {");
+            expect(fixedSourceContents).to.contain(`@use "@blueprintjs/core/lib/scss/variables.scss" as bp;`);
+            expect(fixedSourceContents).to.contain(".#{bp.$bp-ns}-tag {");
```
Comment (styu) [inline packages/stylelint-plugin/test/no-prefix-literal.test.js]:
```suggestion
            expect(fixedSourceContents).to.contain(".#{bp.$ns}-tag {");
```

File: packages/stylelint-plugin/src/utils/checkImportExists.ts [hunk 6]
```diff
@@ -15,17 +15,43 @@
...
+                }
+            } else {
+                const asText = ` as ${namespace}`;
+                if (
+                    atRule.params.endsWith(asText) &&
+                    stripQuotes(atRule.params.substring(0, atRule.params.lastIndexOf(asText))) === path
+                ) {
+                    hasBpVarsImport = true;
+                    return false; // Stop the iteration
+                }
```
Comment (styu) [inline packages/stylelint-plugin/src/utils/checkImportExists.ts]: is this equivalent?
```suggestion
            if (stripQuotes(stripLessReference(atRule.params)) === path) {
                hasBpVarsImport = true;
                return false; // Stop the iteration
            }
            if (namespace !== undefined) {
                const asText = ` as ${namespace}`;
                if (
                    atRule.params.endsWith(asText) &&
                    stripQuotes(atRule.params.substring(0, atRule.params.lastIndexOf(asText))) === path
                ) {
                    hasBpVarsImport = true;
                    return false; // Stop the iteration
                }
```

File: packages/stylelint-plugin/src/rules/no-color-literal.ts [hunk 7]
```diff
@@ -83,10 +84,15 @@ export default stylelint.createPlugin(
             const importPath = options?.variablesImportPath?.[cssSyntaxType] ?? BpVariableImportMap[cssSyntaxType];
             const extension = CssExtensionMap[cssSyntaxType];
             if (hasBpVariablesImport == null) {
-                hasBpVariablesImport = checkImportExists(root, [importPath, `${importPath}.${extension}`]);
+                hasBpVariablesImport = checkImportExists(
+                    cssSyntaxType,
+                    root,
+                    [importPath, `${importPath}.${extension}`],
```
Comment (adidahiya) [inline packages/stylelint-plugin/src/rules/no-color-literal.ts:90]: minor: can we now infer the file extension from the first argument `cssSyntaxType`? it seems like we no longer need an array of import paths, we can just provide one and `checkImportExists` should check for both paths with and without the extension.

**Review by adidahiya — APPROVED** [review]
LGTM other than the one comment here. We'll release this as a breaking change to the rule in a new major version for the plugin.

File: packages/stylelint-plugin/src/rules/no-color-literal.ts (hunk 7 above)
Comment (adidahiya) [reply packages/stylelint-plugin/src/rules/no-color-literal.ts:90]: I'll address this in a follow-up PR

Comment (kmark) [reply packages/stylelint-plugin/src/rules/no-color-literal.ts:90]: https://github.com/palantir/blueprint/pull/5654

**Pull request merged**